### PR TITLE
Show and highlight web apps on OpenMS website

### LINKED ISSUE #136: Show and highlight web apps on OpenMS website
We generate our website with gohugo. Hugo will in general convert markdown .md and restructured text .rst files into html. For custom styling we still need html code sections, which are in our partials folder: https://github.com/OpenMS/OpenMS-website/tree/main/layouts/partials

We would need some nice way to present our web apps on the website, with mainly a name and some kind of logo. We can also remove the overview of proteomics, metabolomics and workflows, which gets its structrur from hero items in the partials folder. I would recommend looking at examples from different website that also use gohugo how to create a html partial https://gohugo.io/showcase/. Because we want to add more apps in time, we probably need some scroll or search function for the apps.

We can also make use of the hero item partial and change it.

### PR DESCRIPTION
**Pull request by JKVISION1101**
#### Brief description of what is fixed or changed

fixes #136 

Describe your changes here
--> 
Highlighted Web Apps on the OpenMS Website using **carousel** style.: Enhanced visibility and accessibility of the web applications available on the OpenMS website by highlighting them prominently.

___

### **PR Type**
enhancement, documentation

___

### **Description**
- Integrated a new carousel component using Swiper.js to highlight web applications on the OpenMS website.
- Added Bootstrap and jQuery libraries to enhance UI and functionality.
- Created a new section for web apps with dynamic loading from configuration.
- Updated styles and scripts to support the new carousel and web apps display.
- Simplified the key features section layout by removing feature box content.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>baseof.html</strong><dd><code>Integrate Bootstrap and Swiper.js for enhanced UI</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

layouts/_default/baseof.html

<li>Added Bootstrap and Swiper.js libraries.<br> <li> Integrated jQuery for enhanced functionality.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-ec6e7cb6e89cbe10d49085811bd67ad23df375247295fc387da90f5fdc56de7f">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.html</strong><dd><code>Add web apps carousel to homepage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

layouts/index.html

- Added partial for web apps carousel.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-96762550561f76f989ced99c4a0751e94696682490ee94b2d77a66377c9619eb">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>carousel.html</strong><dd><code>Create carousel component for web apps</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

layouts/partials/carousel.html

<li>Created a new carousel component with multiple items.<br> <li> Each item includes a card with an image, title, text, and link.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-ec5b772500e55b934612e31d65c13d2a85a1ca18772afb0add84808dfdd8c211">+111/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>css.html</strong><dd><code>Include Swiper CSS for carousel styling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

layouts/partials/css.html

- Added Swiper CSS for carousel styling.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-e667f836a02d67217a1cf419d21974e3ff9945d82235e2310033f9c3e6dd45f6">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>keyfeatures.html</strong><dd><code>Simplify key features section layout</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

layouts/partials/keyfeatures.html

- Removed feature box content.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-6d86986970fb100aa7691dfe90c64eede9c787efbf649134cc9c03dcb997fa02">+1/-11</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>webapps.html</strong><dd><code>Add dynamic web apps section with Swiper</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

layouts/partials/webapps.html

<li>Added new section for web apps with Swiper integration.<br> <li> Dynamically loads web apps from configuration.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-d41c7f7aa4e3f2ed313d36350e70f2dd54f7d1750c562b54aaf0222871ae12a7">+22/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>webapps_1.html</strong><dd><code>Add alternative web apps display with card layout</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

layouts/partials/webapps_1.html

<li>Added alternative web apps display with card layout.<br> <li> Included styling for card interaction.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-686944556e47ade7d650bf26a25b13ed6a7ef913f14230e193b2129903f23d06">+80/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>webapps.css</strong><dd><code>Style web apps container and Swiper slides</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

assets/css/webapps.css

<li>Added styles for web apps container and Swiper slides.<br> <li> Defined styles for slide images and buttons.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-e939772f2afbebdb74e2c15282868d989a18c1c1668bf0c1014b10c3206590b5">+87/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>swiper.js</strong><dd><code>Initialize Swiper and add slide interactions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

assets/js/swiper.js

<li>Initialized Swiper with navigation and pagination.<br> <li> Added event listeners for slide interactions.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-001e68775dfec92fda4d7c021564d6b51d67da350055e23cabb4f11145dc24be">+80/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>config.yaml</strong><dd><code>Configure web apps with URLs and descriptions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

config.yaml

- Added configuration for web apps including URLs and descriptions.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS-website/pull/162/files#diff-d8d0422389f03d783e32e627250fe29834bd09c6361640d1ff00661dd6820034">+34/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: layouts/_default/baseof.html [hunk 1]
```diff
@@ -57,5 +58,8 @@
     {{ template "_internal/google_analytics_async.html" . }}
 
     {{ partial "analytics.html" . }}
+    <script src="https://code.jquery.com/jquery-3.7.1.min.js" integrity="sha256-/JqT3SQfawRcv/BIHPThkBvs0OEvtFFmqPF/lYI/Cxo=" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/swiper@11/swiper-bundle.min.js"></script>
```
Comment (jpfeuffer) [inline layouts/_default/baseof.html:63]: I think we should decide. Either package and ship it. Or load it from a cdn.
See the swiper.js under assets.

File: layouts/partials/carousel.html
Comment (jpfeuffer) [inline layouts/partials/carousel.html:1]: Is this file still needed?

File: layouts/index.html [hunk 2]
```diff
@@ -18,6 +18,9 @@
     {{ if .Site.Params.keyfeatures }}
     {{ partial "keyfeatures.html" . }}
     {{ end }}
+
+    {{partial "webapps.html" . }}
+    <!--{{partial "webapps_1.html" . }}-->
```
Comment (jpfeuffer) [inline layouts/index.html:23]: Remove?

File: layouts/partials/webapps_1.html
Comment (jpfeuffer) [inline layouts/partials/webapps_1.html:1]: Remove?

File: static/images/webapp/logo/clickhere.png
Comment (jpfeuffer) [inline static/images/webapp/logo/clickhere.png:1]: Remove?

**Comment by jpfeuffer** [comment]
Oh, and by the way, can you check the behaviour on mobile? The slider is way too big there.

File: layouts/index.html (hunk 2 above)
Comment (JKVISION1101) [reply layouts/index.html:23]: Right, I will remove it.

File: layouts/partials/webapps_1.html
Comment (JKVISION1101) [reply layouts/partials/webapps_1.html:1]: Thanks for comment. I will remove it.

File: static/images/webapp/logo/clickhere.png
Comment (JKVISION1101) [reply static/images/webapp/logo/clickhere.png:1]: I will remove it. Thanks for comment.

**Comment by JKVISION1101** [comment]
@jpfeuffer 

> Oh, and by the way, can you check the behaviour on mobile? The slider is way too big there.

Yes,  I checked it. it is problem.  I will fix it. Thanks for comment.

**Comment by JKVISION1101** [comment]
@jpfeuffer @matteopilz 
Our team made new pull request because of conflict issue. #165

**Pull request closed without merging**